tools/stats-exporter/main.go

```diff
@@ -0,0 +1,132 @@
+// needed to execute the Query commands. In unit tests, we can moock this
+// function and return the dbQueryable type and eliminate the need for having
+// a live database up when tests run or mocking the rows
+var sqlOpen = func(driver, dsn string) (dbQueryable, error) {
+	return sql.Open(driver, dsn)
+}
+
+// Used to to enable unit tests where we don't want to actually run commands
+// on the host. Instead, we can mock the cmd.Run functions and focus on the
+// error logic
+var execRun = func(c *exec.Cmd) error {
+	return c.Run()
+}
+
+// Connect to the database and run the select query
+func queryDB(dbConnect, beginTimeStamp, endTimeStamp string) (*sql.Rows, error) {
+	dbDSN, err := ioutil.ReadFile(dbConnect)
+	if err != nil {
+		return nil, fmt.Errorf("Could not open database connection file %q: %s", dbConnect, err)
+	}
+	db, err := sqlOpen("mysql", strings.TrimSpace(string(dbDSN)))
+	if err != nil {
+		return nil, fmt.Errorf("Could not establish database connection: %s", err)
+	}
+	rows, err := db.Query(`SELECT id,reversedName,notBefore,serial FROM issuedNames where notBefore >= ? and notBefore < ?`, beginTimeStamp, endTimeStamp)
+	if err != nil {
+		return nil, fmt.Errorf("Could not complete database query: %s", err)
+	}
+	return rows, nil
+}
+
+// Write the query results in TSV format
+func writeTSVData(rows sqlRows, outFile io.Writer) error {
+	for rows.Next() {
+		var (
+			id, rname, notBefore, serial string
+		)
+		if err := rows.Scan(&id, &rname, &notBefore, &serial); err != nil {
+			return err
+		}
+		if _, err := fmt.Fprintf(outFile, "%s\t%s\t%s\t%s\n", id, rname, notBefore, serial); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Compress the results TSV file
+func compress(outputFileName string) error {
+	gzipCmd := exec.Command("gzip", outputFileName)
+	if err := execRun(gzipCmd); err != nil {
+		return fmt.Errorf("Could not gzip result file: %s", err)
+	}
+	return nil
+}
+
+// SCP the compressed file to a remote host
+func scp(outputFileName, destination, key string) error {
+	outputGZIPName := outputFileName + ".gz"
+	scpCmd := exec.Command("scp", "-i", key, outputGZIPName, destination)
+	if err := execRun(scpCmd); err != nil {
+		return fmt.Errorf("Could not scp result file %q to %q: %s", outputFileName, destination, err)
+	}
+	return nil
+}
+
+func main() {
+	dbConnect := flag.String("dbConnect", "", "Path to the DB URL file")
+	destination := flag.String("destination", "localhost:/tmp", "Location to SCP the TSV result file to")
+	key := flag.String("key", "id_rsa", "Identity key for SCP")
+	flag.Parse()
+
+	// The query we run against the database is to examine the previous day of data
+	// we construct dates that correspond ot the start and stop of theat 24 hour window
+	now := time.Now()
+	yesterday := now.Add(-24 * time.Hour)
+	yesterdayDateStamp := yesterday.Format("2006-01-02")
+	endDateStamp := now.Format("2006-01-02")
+	outputFileName := fmt.Sprintf("results-%s.tsv", yesterday.Format("2006-01-02"))
```

Comment:
I think you can use `yesterdayDateStamp` as the second arg here instead of formatting `yesterday` again inline:
```suggestion
	outputFileName := fmt.Sprintf("results-%s.tsv", yesterdayDateStamp)
```